File: ProjectPos.BackEnd/ProjectPos.Services/DTOs/GoodsReceivedVoucherDto.cs
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using ProjectPos.Data.EntityModels;
using ProjectPos.Data.Shared.Enums;
using ProjectPos.Services.EntityDtos;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace ProjectPos.Services.DTOs
{
    public class GoodsReceivedVoucherDto : FullAuditedEntityDto<int>
    {
        public int? OrderNumber { get; set; }
        public string? InvoiceNumber { get; set; }
        public string? Transpoter { get; set; }
        public int? SupplierId { get; set; }
        [Column(TypeName = "double(12, 2)")]
        public double? PaidAmount { get; set; }
        [Column(TypeName = "double(12, 2)")]
        public double? UsdPaidAmount { get; set; }
        [Column(TypeName = "double(12, 2)")]
        public double? AmountDue { get; set; }
        public string? SupplierName { get; set; }

        [Column(TypeName = "double(12, 2)")]
        public double? Value { get; set; }
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public OrderPaymentStatus? Status { get; set; }
        public bool IsPaid { get; set; } = false;
        public bool IsApproved { get; set; } = false;

        [ForeignKey("SupplierId")]
        public CompanyDto? Supplier { get; set; }

        //dependent entities
        public List<GoodsReceivedVoucherLineDto>? ReceivedItems { get; set; }
        public List<PurchaceOrderPaymentsDto>? PurchaceOrderPayments { get; set; }
    }
}

File: ProjectPos.BackOffice.FrontEnd/src/proxy/interfaces/goods-received-voucher-dto.ts
import { FullAuditedEntityDto } from "../entity-dtos/full-audited-entity-dto";
import { OrderPayStatus } from "../enums/order-pay-status";
import { OrderPaymentStatus } from "../enums/order-payment-status";
import { CompanyDto } from "./company-dto";
import { GoodsReceivedVoucherLineDto } from "./goods-received-voucher-line-dto";
import { PurchaceOrderDto } from "./purchace-order-dto";
import { PurchaceOrderPaymentDto } from "./purchace-order-payment-dto";

export interface GoodsReceivedVoucherDto extends FullAuditedEntityDto<number> {
    orderNumber?: number;
    invoiceNumber?: string;
    transpoter?: string;
    supplierId?: number;
    supplierName?: string;
    value?: number;
    isApproved?: boolean;
    paidAmount?: number;
    isPaid?: boolean;
    usdPaidAmount?: number;
    amountDue?: number;
    status?: OrderPaymentStatus;
    purchaseOrder?: PurchaceOrderDto;
    supplier?: CompanyDto;
    receivedItems?: GoodsReceivedVoucherLineDto[];
    purchaceOrderPayments?: PurchaceOrderPaymentDto[];
}
